### PR TITLE
Raw dataset retrieval and static HDF5 dataset generation

### PR DESCRIPTION
**Pull request by michaelkiper**
## Overview
This is a replica of a previous PR that was open but seems to be closed with no trace, likely due to the permissions issues of GitHub around repos.

This PR establishes the retrieval of raw datasets from either a URL or a Unix filesystem path. From the obtained source it will get the CSV, and use the following logic to get the spectra:
- the longest contiguous chunk of column headers that are valid floating points will be used at the spectra data matrix
The reason for using it this ways that A) it creates a column position agnostic code B) aligns with what I have seen from all of the relevant CSVs on the emit hpc VFS. 

Output HDF5s will have the following contents:
1. datasets:
     - "spectra": the torch float32 matrix of the interpolated raw spectra
2. attributes:
     - "wavelengths": the wavelengths of the spectra
     - "original-ds ": Original source endpoint of the data
     - 
No class labels will be provided as the file itself indicates the class, so it's not needed.

There is also a `interior_interpolation` function that interpolates the wavelengths to the 1nm wavelength range. 

## Testing
The testing done for this was to establish a set of unit tests and pass those along with regression tests.

### CODE REVIEW

File: src/cover_class/static/preprocessor.py [hunk 1]
```diff
@@ -1,6 +1,21 @@
-from torch import FloatTensor
+from typing import Tuple
+from torch import FloatTensor, Tensor
+import torch
+from numpy.typing import NDArray
+import numpy as np
+from scipy.interpolate import interp1d # type: ignore[import]
 
-def interpolator(data_matrix:FloatTensor) -> FloatTensor: ... # type: ignore 
+def interior_interpolation(
```
Comment (michaelkiper) [inline src/cover_class/static/preprocessor.py:8]: This is the 1nm interpolation strategy I had in mind - lmk if there is a better implementation.
